src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/ThrowsTaglet.java

```diff
@@ -233,7 +236,8 @@ protected Content throwsTagsOutput(Map<List<? extends ThrowsTree>, ExecutableEle
      * @param instantiatedThrownTypes the thrown types in the context of the current type.
      * @return map of declared to instantiated thrown types or an empty map.
      */
-    private Map<String, TypeMirror> getSubstitutedThrownTypes(List<? extends TypeMirror> declaredThrownTypes,
+    private Map<String, TypeMirror> getSubstitutedThrownTypes(Types types,
```

Comment:
ok, but mild uugh, and I don't see a better way to handle it in the current architecture.

In general, objects like `utils`, `elements, `types` get cached because they are unique for life of the doclet.  But that brings up an initialization issue, since they currently only come from the writer.